### PR TITLE
mongodb : update MongoDBChatMessageHistory to accept existing mongo_client

### PR DESCRIPTION
**Pull request by SuperMuel**
### Description

This PR enhances the `MongoDBChatMessageHistory` class by adding support for initialization with an existing `MongoClient` instance.

#### Benefits:
- **Custom Configurations:** Allows the use of a `MongoClient` instance with specific configurations.
- **Performance Improvement:** Enhances performance by preventing the repeated creation of new `MongoClient` instances for each session, thereby eliminating the need for re-authentication on every new session.

#### Instantiation with a `connection_string` (current method):
```python
from langchain_mongodb import MongoDBChatMessageHistory

history = MongoDBChatMessageHistory(
    connection_string="mongodb://your-host:your-port/",
    session_id="your-session-id",
)
```

#### Instantiation with an existing `MongoClient` (new feature in this PR):
```python
from langchain_mongodb import MongoDBChatMessageHistory
from pymongo import MongoClient

existing_client = MongoClient("mongodb://your-host:your-port/")
history2 = MongoDBChatMessageHistory(
    connection_string=None,
    session_id="your-session-id",
    client=existing_client,
)
```

**Note:**  
Due to the existing parameter order where `connection_string` precedes `session_id`, and since `session_id` does not (and should not) have a default value, I think it’s not possible to set a default value of `None` for `connection_string`. Changing the order of these parameters would introduce breaking changes.

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
cc @Jibola

File: libs/partners/mongodb/langchain_mongodb/chat_message_histories.py [hunk 1]
```diff
@@ -70,12 +70,14 @@ def __init__(
         create_index: bool = True,
         history_size: Optional[int] = None,
         index_kwargs: Optional[Dict] = None,
+        mongo_client: Optional[MongoClient] = None,
```
Comment (efriis) [inline libs/partners/mongodb/langchain_mongodb/chat_message_histories.py:73]:
```suggestion
        client: Optional[MongoClient] = None,
```

File: libs/partners/mongodb/langchain_mongodb/chat_message_histories.py [hunk 2]
```diff
@@ -93,19 +95,28 @@ def __init__(
                 count of (most recent) messages to fetch from MongoDB.
             index_kwargs: Optional[Dict]
                 additional keyword arguments to pass to the index creation.
+            mongo_client: Optional[MongoClient]
```
Comment (efriis) [inline libs/partners/mongodb/langchain_mongodb/chat_message_histories.py:98]:
```suggestion
            client: Optional[MongoClient]
```

File: libs/partners/mongodb/langchain_mongodb/chat_message_histories.py [hunk 3]
```diff
@@ -93,19 +95,28 @@ def __init__(
...
+            self.client = mongo_client
+        elif connection_string:
+            try:
+                self.client = MongoClient(connection_string)
+            except errors.ConnectionFailure as error:
+                logger.error(error)
+        else:
+            raise ValueError(
+                "Either connection_string or mongo_client must be provided"
+            )
```
Comment (efriis) [inline libs/partners/mongodb/langchain_mongodb/chat_message_histories.py:119]:
```suggestion
        if client:
            if connection_string:
                raise ValueError("Must provide connection_string or client, not both")
            self.client = client
        elif connection_string:
            try:
                self.client = MongoClient(connection_string)
            except errors.ConnectionFailure as error:
                logger.error(error)
        else:
            raise ValueError(
                "Either connection_string or client must be provided"
            )
```

**Review by efriis — APPROVED** [review]

**Review by Jibola — CHANGES_REQUESTED** [review]
Overall, LGTM! Thanks for adding it!

Could you add a constructor test for it since it's introducing a new functionality?

**Comment by SuperMuel** [comment]
I implemented some constructor tests, hope it's adequate :) @Jibola

**Comment by SuperMuel** [comment]
Hey, I fix the lints in a few hours

**Comment by SuperMuel** [comment]
@efriis Should be good :)

**Comment by haarisedhi102** [comment]
@SuperMuel nice work, looking forward to this change

**Comment by efriis** [comment]
migrating this to the new partner repo in langchain-ai/langchain-mongodb - sorry for the delay!

**Pull request closed without merging**